HLX503/iocBoot/iocHLX503-IOC-01/st-common.cmd

```diff
@@ -39,7 +39,7 @@ $(IFNOTDEVSIM) $(IFNOTRECSIM) asynSetOption("$(DEVICE)",0,"crtscts","N")
 ## Software flow control off
 $(IFNOTDEVSIM) $(IFNOTRECSIM) asynSetOption("$(DEVICE)",0,"ixon","N")
 $(IFNOTDEVSIM) $(IFNOTRECSIM) asynSetOption("$(DEVICE)",0,"ixoff","N")
-$(IFNOTDEVSIM) $(IFNOTRECSIM) #asynInterposeThrottleConfig("L0", -1, 0.1)
+$(IFNOTDEVSIM) $(IFNOTRECSIM) asynInterposeThrottleConfig("L0", -1, 0.1)
```

Comment:
Just looking briefly at what is scanning in the ITC DB which gets loaded, I can see 12 items scanning at `1 second`. Not sure I understand how this works with the asynInterposeThrottle enforcing 0.1s between each command? (This is before taking into account any extra writes from the user or the statemachine).

If there's some subtelty of this asynThrottleConfig I've missed please document in a comment/on the wiki

Reply:
The throttling was for the state machine rather than using pvput(..., SYNC) on each call - seemed to fix some recondense steps as well as channel switching to avoid races. The ITC doesn't seem to keep up if polling as well as doing a sequence of writes from a SWITCH_CHANNEL:SP otherwise!

Reply:
sure - but it seems to me like 12 commands each scanning at 1s and at a minimum separation of 0.1s shouldn't work, even before taking into account the writes from the state machine? Maybe best to discuss in a call if you like?

Reply:
`asynInterposeThrottleConfig` sets a min delay between writes on the port, so as you say if you were to have more than 10 records scanning at 1 second interval they would not all complete in 1 second. What i believe then happens is the next 1 second scan will see some records still "busy" as they had queued a scan with asyn but that had not yet happened. In this situation it will check the record lock count field LCNT and if this is less than 10 it will increment it and not scan the record that time. So though you will get 12 record writes queued first time, next time it will only queue 10 as 2 will still be "busy" etc. If a record is found to be "busy" ten consecutive times, it gets put into SCAN alarm. So i think epics will even things out for you, but ideally we would want things to fully complete in the scan.  Does everything need to scan that quickly?

Reply:
I think we have a spare ITC we can test with from r106 as a dummy to see if it falls over with 20-50ms - will report back